fedimint-server/src/consensus/mod.rs

```diff
@@ -530,6 +530,16 @@ impl FedimintConsensus {
 
         let tx_hash = transaction.tx_hash();
 
+        // Prevent transaction replays
+        if dbtx
+            .get_value(&AcceptedTransactionKey(tx_hash))
+            .await
+            .expect("DB error")
+            .is_some()
+        {
+            return Err(TransactionSubmissionError::TransactionReplayError(tx_hash));
```

Comment:
Should it be an error? Maybe just an OK and do nothing? In that case the call is idempotent.

Reply:
I wasn't aware we are already storing all accepted txes.

Reply:
> Should it be an error? Maybe just an OK and do nothing? In that case the call is idempotent.

I think that would mess with your recovery code since the tx would not be marked as invalid in the epoch history.

The alternative I see is checking outside this fn, right before it is called. That way we don't write a tx failed to the DB but can still mark it as invalid in the epoch history. That actually sounds better to me.

Reply:
Oh. This is already on processing consensus. OK then.

I was thinking that we could do this check on submission and then return `Ok`.

Reply:
Right, on submission we want a similar check, that's part of #1353. But in case guardians are malicious we definitely need the consensus check here.